pkgpanda/http/__init__.py

```diff
@@ -0,0 +1,229 @@
+"""Pkgpanda HTTP API"""
+
+import http.client
+import logging
+import os
+import sys
+
+from flask import Flask, current_app, jsonify, request
+
+from pkgpanda import Install, Repository, actions
+from pkgpanda.exceptions import (PackageConflict, PackageError,
+                                 PackageNotFound, ValidationError)
+
+
+empty_response = ('', http.client.NO_CONTENT)
+
+
+def package_listing_response(package_ids):
+    return jsonify(sorted(package_ids))
+
+
+def package_response(package_id, repository):
+    try:
+        package = repository.load(package_id)
+    except ValidationError:
+        response = (
+            error_response("Invalid package ID: {}".format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageNotFound:
+        response = (
+            error_response('Package {} not found.'.format(package_id)),
+            http.client.NOT_FOUND,
+        )
+    except PackageError:
+        error_message = 'Unable to load package {}.'.format(package_id)
+        logging.exception(error_message)
+        response = (
+            error_response(error_message),
+            http.client.INTERNAL_SERVER_ERROR,
+        )
```

Comment:
I think adding HTTP error responses to exception classes would be a needless coupling. API error responses are relevant only to this module, so they probably ought to be defined here, not as part of exception classes that aren't otherwise HTTP-aware. And we can't assume a 1:1 relationship between exception classes and error responses -- the error response may be different based on context that's only available to the caller. For example there are a couple cases here where we send a different error in response to `ValidationError` and `PackageError` based on context.

👍  on namedtuple responses, I'll add that.